src/scan/postgres_scan.cpp

```diff
@@ -44,11 +44,11 @@ PostgresScanGlobalState::InitGlobalState(duckdb::TableFunctionInitInput &input)
 
 	if (input.CanRemoveFilterColumns()) {
```

Comment:
I think adding some comments to this section would be very helpful. It's not super clear to me what the differences between these branches is (primarily because the comments in duckdb itself on the relevant functions and fields are non-existing).